CHROMEOS Modify LAVA template: add modules into rootfs

nuclearcat opened the pull request:
LAVA capable to add upstream modules into ChromiumOS rootfs image
using libguestfs, and this way we can test upstream kernels.

First tests:
"Fixed kernel" https://lava.collabora.co.uk/scheduler/job/6015601
Upstream kernel https://lava.collabora.co.uk/scheduler/job/6015641

Signed-off-by: Denys Fedoryshchenko <denys.f@collabora.com>

gctucker commented:
> First tests: "Fixed kernel" https://lava.collabora.co.uk/scheduler/job/6015601

I don't see the modules being installed in this job. Was it just using the built-in ones instead?

nuclearcat commented:
Yes, fixed kernel is to use rootfs image "as is", it is affected by template chane as well so i retested it.

I can test also with "known good" kernel and modules from newer "ChromiumOS" specific kernel build.

gctucker commented:
We could have a fixed kernel with an image and a modules tarball hosted on storage.chromeos.kernelci.org, just it won't be a testing one built by KernelCI.

nuclearcat commented:
Can we use https://storage.chromeos.kernelci.org/images/kernel/chromeos-x86-5.15.15/ ?

nuclearcat commented:
I found more elegant way of solving bzImage "lowercasing" problem.
I suspect it is happening, because this url (downloads://bzImage) parsing bzImage as hostname/netloc, which is by definition case insensitive.
So specifying downloads://./bzImage fixes issue.

nuclearcat commented:
Example of job: https://lava.collabora.co.uk/scheduler/job/6049419

gctucker commented on config/lava/chromeos/chromiumosvm-tast-template.jinja2:
```diff
@@ -7,6 +7,35 @@
+             compression: xz
+    postprocess:
+        docker:
+            image: denysfcollabora/lavapost
```
This will need to be replaced with the permanent `kernelci/` image from #1110

gctucker reviewed (commented):
Looks good. How about storing a Chrome OS "fixed" kernel with modules so we could test this with a working kernel and confirm it's really booting as expected?  If we had that, we could even just use hard-coded URLs for the kernel and modules instead of built-in binaries with the `fixed_kernel` parameter.

nuclearcat commented:
> Looks good. How about storing a Chrome OS "fixed" kernel with modules so we could test this with a working kernel and confirm it's really booting as expected? If we had that, we could even just use hard-coded URLs for the kernel and modules instead of built-in binaries with the `fixed_kernel` parameter.

I will investigate that. For tests i was extracting modules and kernel manually from rootfs image, but i think i can automate this extraction from SDK during rootfs build process and can put them in same directory where we put rootfs and tast tgz. 
Is it ok like that?

gctucker commented:
> > Looks good. How about storing a Chrome OS "fixed" kernel with modules so we could test this with a working kernel and confirm it's really booting as expected? If we had that, we could even just use hard-coded URLs for the kernel and modules instead of built-in binaries with the `fixed_kernel` parameter.
> 
> I will investigate that. For tests i was extracting modules and kernel manually from rootfs image, but i think i can automate this extraction from SDK during rootfs build process and can put them in same directory where we put rootfs and tast tgz. Is it ok like that?

I guess it would be good to have an automated way of doing this to be able to reproduce it, but the bare minimum at this stage is to have a known good kernel we can use as a reference.  So if it's easy to add the automated way then that sounds good, otherwise you could do it by hand now so we can use it to test the on-going changes and automate this later.

nuclearcat commented:
Docker image name updated to proper one

gctucker commented on config/core/test-configs.yaml:
```diff
@@ -189,6 +189,17 @@ test_plans:
+    params:
+      job_timeout: '60'
+      docker_image: 'kernelci/chromeos-tast'
+      rootfs_url: 'https://storage.chromeos.kernelci.org/images/rootfs/chromeos/crostest006/chromiumos-amd64-generic/amd64/'
```
It looks like this wasn't updated using the main rootfs-builder job but one you ran with your branch, as the "pipeline version" is not a date like `20220412.0`.  Maybe that's OK but it's better to be sure it was really built with the `chromeos.kernelci.org` branch so we know it can be reproduced easily.

gctucker replied:
This still hasn't been fixed, the main Chrome OS rootfs build job should have uploaded it to a URL with the current date.

gctucker commented on config/core/test-configs.yaml:
```diff
@@ -189,6 +189,17 @@ test_plans:
+    params:
+      job_timeout: '60'
+      docker_image: 'kernelci/chromeos-tast'
+      rootfs_url: 'https://storage.chromeos.kernelci.org/images/rootfs/chromeos/crostest006/chromiumos-amd64-generic/amd64/'
```
It looks like this wasn't updated using the main rootfs-builder job but one you ran with your branch, as the "pipeline version" is not a date like `20220412.0`.  Maybe that's OK but it's better to be sure it was really built with the `chromeos.kernelci.org` branch so we know it can be reproduced easily.

gctucker commented:
FYI Tests on chromeos.kernelci.org haven't run for about a week now, so that needs to be fixed if we want to fully test pending PRs
https://chromeos.kernelci.org/job/next/branch/master/

gctucker commented:
FYI Tests on chromeos.kernelci.org haven't run for about a week now, so that needs to be fixed if we want to fully test pending PRs
https://chromeos.kernelci.org/job/next/branch/master/

nuclearcat commented:
I will investigate this today

nuclearcat commented:
Issue is fixed, it is due https://kernelci.org/docs/instances/production/#known-issues , thanks for @mgalka .

gctucker commented:
> Issue is fixed, it is due https://kernelci.org/docs/instances/production/#known-issues , thanks for @mgalka .

No, it's not that.

nuclearcat commented:
I am sure i added it last time, but right now i checked, schedule was empty again. I just added it back now:
![image](https://user-images.githubusercontent.com/1076310/163950487-f5bb8653-0e15-422a-a3e2-c44fa17d2922.png)

gctucker commented:
No that's not how it works. The jobs on chromeos.kernelci.org are triggered once a day with a systemd timer like the staging jobs.

gctucker commented:
> I am sure i added it last time, but right now i checked, schedule was empty again. I just added it back now: ![image](https://user-images.githubusercontent.com/1076310/163950487-f5bb8653-0e15-422a-a3e2-c44fa17d2922.png)

@nuclearcat Where did you enable this?

nuclearcat commented:
As suggested in https://kernelci.org/docs/instances/production/#known-issues - in chromeos/kernel-tree-monitor

gctucker commented:
OK it got disabled again automatically by the regular DSL job, so that's fine.

Someone still needs to investigate the issue with the chromeos.kernelci.org jobs.  It's run by the script in kernelci-deploy, just like staging.

nuclearcat commented:
I guess @mgalka can help when he have time? As i guess it require sysadmin privileges.

gctucker commented:
I don't think sysadmin privilege is actually needed, most likely there's a problem with patches on the chromeos.kernelci.org branch.  But yes @mgalka can help with that while I'm fixing production Jenkins today.

nuclearcat commented:
Rebased

nuclearcat commented:
As far as i see it works over jenkins:
https://chromeos.kernelci.org/test/job/next/branch/master/kernel/next-20220422/plan/chromiumosvm-tast-upstream-kernel/

https://storage.chromeos.kernelci.org/next/master/next-20220422/x86_64/x86_64_defconfig+x86-chromebook/clang-13/lab-collabora/chromiumosvm-tast-upstream-kernel-qemu_x86_64-uefi-chromiumos.txt

nuclearcat commented:
URL of rootfs updated, waiting test results after next trigger

nuclearcat commented:
Found mistake (copy&paste issue), new revision, tested manually at: https://lava.collabora.co.uk/scheduler/job/6234225

gctucker commented:
OK thanks for testing this manually. I'll merge it now as I think it's fine but we should confirm we get some actual results on chromeos.kernelci.org.

gctucker commented:
Also I'll do a rebase of the `chromeos` branch now and tidy things up a bit.
